lib/src/messaging_client.dart

```diff
@@ -0,0 +1,159 @@
+
+import 'package:flutter_aira/src/models/sent_file_info.dart';
+import 'package:flutter_aira/src/platform_client.dart';
+import 'package:flutter_aira/src/platform_exceptions.dart';
+import 'package:logging/logging.dart';
+import 'package:pubnub/pubnub.dart' as pn;
+
+import 'models/message.dart';
+import 'models/session.dart';
+
+/// Room independent class to handle all messaging.
+abstract class MessagingClient {
+  /// The ID of the service request.
+  set serviceRequestId(int id);
+
+  /// A broadcast stream of messages sent and received.
+  ///
+  /// If the application does not support messaging, this will throw an exception.
+  Stream<Message> get messageStream;
+
+  Future<void> sendStart();
```

Comment:
I forgot to add a comment here... will do! This function, it turns out, is essential for fixing the issue where messages from previous call sneaks into later calls. It is only used by this sdk though and I would like to hide this function from the API. I left this function here for now as I didn't want to delay implementation of the feature.